src/output.c

```diff
@@ -45,12 +45,12 @@ void open_output(char *filename)
     int fd;
     int opts
         = (O_CREAT
-           | (seek_records || (conversions_mask & C_NOTRUNC) ? 0 : O_TRUNC));
+           | ((seek_records || do_diffwr || (conversions_mask & C_NOTRUNC)) ? 0 : O_TRUNC));
 
     /* Open the output file with *read* access only if we might
-       need to read to satisfy a `seek=' request.  If we can't read
+       need to read to satisfy a `seek=' or "diffwr=on" request. If we can't read
```

Comment:
This is only a minor cosmetic issue, but to maintain consistency, ``` `diffwr=on'``` (starting with backquote and ending with single quote) should be used instead.